### PR TITLE
Emit debug locations for hop_to_executor instructions.

### PR DESCRIPTION
**Pull request by adrian-prantl**
The function prologue of async funclets inherits its source location
from the hop_to_executor instruction. This makes it easier to produce
logical backtraces, since the PC in logical frames will always point
to the start if the function.

rdar://89776340

### CODE REVIEW

**Comment by adrian-prantl** [comment]
@swift-ci test and merge

**Comment by adrian-prantl** [comment]
@swift-ci test

**Comment by adrian-prantl** [comment]
@swift-ci test and merge

**Comment by adrian-prantl** [comment]
@swift-ci test and merge

**Comment by adrian-prantl** [comment]
@swift-ci smoke test

File: include/swift/SIL/SILLocation.h [hunk 1]
```diff
@@ -344,12 +344,13 @@ class SILLocation {
 
   /// Returns true if the location represents an artificially generated
   /// body, such as thunks or default destructors.
-  ///
-  /// These locations should not be included in the debug line table.
-  /// These might also need special handling by the debugger since they might
-  /// contain calls, which the debugger could be able to step into.
   bool isAutoGenerated() const { return kindAndFlags.fields.autoGenerated; }
 
+  /// Returns true if the location should be represented in debuginfo.
```
Comment (eeckstein) [inline include/swift/SIL/SILLocation.h:349]: `should` -> `shouldn't` ?

**Review by eeckstein — APPROVED** [review]
lgtm!

**Comment by adrian-prantl** [comment]
@swift-ci smoke test

**Pull request merged**